src/compiler/type_check.cc

```diff
@@ -431,9 +498,13 @@ class TypeChecker : public ReturningVisitor<Type> {
   }
 
   Type visit_Return(Return* node) {
-    auto return_type = visit(node->value());
+    auto value_type = visit(node->value());
+    return visit_Return(node, value_type);
+  }
+
+  Type visit_Return(Return* node, Type value_type) {
```

Comment:
Are there other callers (not just the `visit_Return(Return*)` above) of this already?

Reply:
not anymore.
Initially I made globals still call the `visit_Return` but I removed that.
reverted.